src/ast/analysis/Functor.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * Souffle - A Datalog Compiler
+ * Copyright (c) 2020 The Souffle Developers. All rights reserved
+ * Licensed under the Universal Permissive License v 1.0 as shown at:
+ * - https://opensource.org/licenses/UPL
+ * - <souffle root>/licenses/SOUFFLE-UPL.txt
+ */
+
+/************************************************************************
+ *
+ * @file Functor.h
+ *
+ * Analysis that provides type information for functors
+ *
+ ***********************************************************************/
+
+#pragma once
+
+#include "ast/IntrinsicFunctor.h"
+#include "ast/Program.h"
+#include "ast/UserDefinedFunctor.h"
+#include "ast/analysis/Analysis.h"
+#include "ast/utility/Visitor.h"
+#include <memory>
+#include <sstream>
```

Comment:
This isn't used. Did you mean `ostream`?

Reply:
Yes ^^